crates/aptos/src/move_tool/mod.rs

```diff
@@ -196,3 +207,86 @@ async fn submit_transaction(
 
     Ok(response.inner().clone())
 }
+
+/// Run a Move function
+#[derive(Parser)]
+pub struct RunFunction {
+    #[clap(flatten)]
+    encoding_options: EncodingOptions,
+    #[clap(flatten)]
+    write_options: WriteTransactionOptions,
+    /// Function name as `<ADDRESS>::<MODULE_ID>::<FUNCTION_NAME>`
+    ///
+    /// Example: `0x842ed41fad9640a2ad08fdd7d3e4f7f505319aac7d67e1c0dd6a7cce8732c7e3::Message::set_message`
+    #[clap(long, parse(try_from_str = parse_function_name))]
+    function_id: FunctionId,
+    /// Hex encoded arguments separated by spaces.
+    ///
+    /// Example: `0x01 0x02 0x03`
+    #[clap(long, multiple_values = true)]
+    args: Vec<HexEncodedBytes>,
+    /// TypeTag arguments separated by spaces.
+    ///
+    /// Example: `u8 u64 u128 bool address vector true false signer`
+    #[clap(long, multiple_values = true)]
+    type_args: Vec<MoveType>,
+}
+
+impl RunFunction {
+    pub async fn execute(self) -> Result<Transaction, CliError> {
+        let args: Vec<Vec<u8>> = self
+            .args
+            .iter()
+            .cloned()
+            .map(|bytes| bytes.into())
```

Comment:
these aren't bytes, they are bcs, which is similar but different enough to confuse the hell out of you